File: 程序员面试金典/4.2最小高度树/sortedArrayToBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return createTree(nums, 0, nums.length);
    }

    private TreeNode createTree(int[] nums, int left, int right) {
        if (left >= right)
            return null;
        int m = (left + right) / 2;
        TreeNode root = new TreeNode(nums[m]);
        root.left = createTree(nums, left, m);
        root.right = createTree(nums, m + 1, right);
        return root;
    }

}

File: 程序员面试金典/4.2最小高度树/sortedArrayToBST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        return self.create_tree(nums, 0, len(nums))
    
    def create_tree(self, nums, left, right):
        if left >= right:
            return None
        m = (left + right) // 2
        root = TreeNode(nums[m])
        root.left = self.create_tree(nums, left, m)
        root.right = self.create_tree(nums, m + 1, right)
        return root
